### PR TITLE
IDE lint: DeckPicker

### PR DESCRIPTION
**Pull request by david-allison**
## Purpose / Description
IDE lint slow down commits. It's best to remove them. Broken window theory applies here: if there are always warnings, Developers are incentivised to skip the check and not check that they've introduced.

I've hit this one a few times today

## Approach
* Autofix IDE Lint issues

`displayDowngradeFailedNoSpace` was unused, which removed `DeckPickerNoSpaceToDowngradeDialog`

We're not going to implement automated backups, so no need for manual backup strings

## Issue
- https://github.com/ankidroid/Anki-Android/issues/13282

## How Has This Been Tested?
⚠️Untested

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 1]
```diff
@@ -2917,42 +2907,18 @@ open class DeckPicker :
         }
     }
 
-    /**
-     * Last warning, asking the user whether they accept risk of data loss.
-     */
-    fun warnAboutBackup() {
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:2923]: I don't understand the removal of this one and the postponement stuff - I thought postponement and risk of data loss were features that were part of scoped storage migration flows?

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:2923]: I'll update the Figma today

Given we're no longer going to implement performing automated backups via `ACTION_OPEN_DOCUMENT_TREE` for the user in 2.16 (proposed by you in Discord AFAIK), then we no longer have a concept of "Manual backups" which this dialog is referring to.

We won't migrate anyone who doesn't sync. Therefore we have no risk of data loss for existing users

For legacy unsafe & new users,
we will warn + prompt for backups, which can be restored from if the user uninstalls

We may also redirect to a non-Play Store build if a legacy user reinstalls.

The "reinstall" case is still unpleasant, but it's much better than migrating and losing data.

Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:2923]: Ah ok - all makes sense when laid out as one coherent thought process

**Review by mikehardy — APPROVED** [review]

**Review by BrayanDSO — APPROVED** [review]
Time to lawn some code

giphy

**Comment by mikehardy** [comment]
strings cleared out

**Pull request merged**